python/ptest/cases/base.py

```diff
@@ -16,6 +16,9 @@ def setup_case(self, simulation):
     def run_case(self, simulation):
         raise NotImplementedError
 
+    def cycle(self, simulation):
+        simulation.flight_controller.write_state("cycle.start", "true")
```

Comment:
You could use `self.sim` instead of `simulation` and this makes the syntax marginally cleaner, since you don't have to pass in simulation as an argument